cypress/e2e/title-level-request/user-can-move-title-level-request-from-one-item-to-another.cy.js

```diff
@@ -0,0 +1,166 @@
+import InventoryInstances from '../../support/fragments/inventory/inventoryInstances';
+import PatronGroups from '../../support/fragments/settings/users/patronGroups';
+import Location from '../../support/fragments/settings/tenant/locations/newLocation';
+import Users from '../../support/fragments/users/users';
+import ServicePoints from '../../support/fragments/settings/tenant/servicePoints/servicePoints';
+import getRandomPostfix, { getTestEntityValue } from '../../support/utils/stringTools';
+import SettingsMenu from '../../support/fragments/settingsMenu';
+import TitleLevelRequests from '../../support/fragments/settings/circulation/titleLevelRequests';
+import Requests from '../../support/fragments/requests/requests';
+import RequestDetail from '../../support/fragments/requests/requestDetail';
+import MoveRequest from '../../support/fragments/requests/move-request';
+
+describe('Title Level Request', () => {
+  let userData = {};
+  const patronGroup = {
+    name: getTestEntityValue('groupTLR'),
+  };
+  const instanceData = {
+    title: getTestEntityValue('InstanceTLR'),
+  };
+  const testData = {
+    userServicePoint: ServicePoints.getDefaultServicePointWithPickUpLocation(),
+  };
+
+  before('Preconditions', () => {
+    cy.getAdminToken()
+      .then(() => {
+        cy.loginAsAdmin({
+          path: SettingsMenu.circulationTitleLevelRequestsPath,
+          waiter: TitleLevelRequests.waitLoading,
+        });
+        ServicePoints.createViaApi(testData.userServicePoint);
+        testData.defaultLocation = Location.getDefaultLocation(testData.userServicePoint.id);
+        Location.createViaApi(testData.defaultLocation);
+        cy.getInstanceTypes({ limit: 1 }).then((instanceTypes) => {
+          testData.instanceTypeId = instanceTypes[0].id;
+        });
+        cy.getHoldingTypes({ limit: 1 }).then((holdingTypes) => {
+          testData.holdingTypeId = holdingTypes[0].id;
+        });
+        cy.createLoanType({
+          name: `type_${getRandomPostfix()}`,
+        }).then((loanType) => {
+          testData.loanTypeId = loanType.id;
+        });
+        cy.getMaterialTypes({ limit: 1 }).then((materialTypes) => {
+          testData.materialTypeId = materialTypes.id;
+        });
+      })
+      .then(() => {
+        instanceData.itemsData = [
+          {
+            barcode: generateUniqueItemBarcodeWithShift(),
+            status: { name: ITEM_STATUS_NAMES.AVAILABLE },
+            permanentLoanType: { id: testData.loanTypeId },
+            materialType: { id: testData.materialTypeId },
+          },
+          {
+            barcode: generateUniqueItemBarcodeWithShift(),
+            status: { name: ITEM_STATUS_NAMES.AVAILABLE },
+            permanentLoanType: { id: testData.loanTypeId },
+            materialType: { id: testData.materialTypeId },
+          },
+        ];
+        InventoryInstances.createFolioInstanceViaApi({
```

Comment:
Please use 'createFolioInstancesViaApi' method instead of this one. New method performs all required preconditions.

You can see it's usage in the following test: cypress/e2e/fee-fine/closed-fee-fines.cy.js

Reply:
this method just create instance with one item, but our requirement about create instance with two or more item, please help me confirm.

Reply:
You should modify generateFolioInstances  method and pass barcodes as parameter:

```JS
  generateFolioInstances({
    count = 1,
    barcodes,
    status = ITEM_STATUS_NAMES.AVAILABLE,
    properties = {},
  } = {}) {
    return [...Array(count).keys()].map((index) => ({
      instanceId: uuid(),
      instanceTitle: `Instance-${getRandomPostfix()}`,
      barcodes: barcodes || [generateUniqueItemBarcodeWithShift(index)],
      status,
      properties: Array.isArray(properties) ? properties[index] : properties,
    }));
  },
```

Then call generateFolioInstances method like this:
```JS
InventoryInstances.generateFolioInstances({
      barcodes: [generateUniqueItemBarcodeWithShift(1), generateUniqueItemBarcodeWithShift(2)]
    }),
```

The next step is to create Instance with 2 items like this:
```JS
        ServicePoints.createViaApi(testData.userServicePoint);
        testData.defaultLocation = Location.getDefaultLocation(testData.userServicePoint.id);
        Location.createViaApi(testData.defaultLocation).then((location) => {
          InventoryInstances.createFolioInstancesViaApi({
            folioInstances: testData.folioInstances,
            location,
          });
        });
```

Reply:
update the code according to your solution,thanks